specification/taf-spec.md

```diff
@@ -0,0 +1,915 @@
+
+Updating metadata and target files in TAF is done in accordance with the TUF specification.
+However, after each update, changes are committed. An update does not simply mean modification
+of any metadata or target file, but a valid update of the TUF repository as a whole. Therefore:
+
+- If any metadata file is updated, it needs to be signed by at least the threshold of keys.
+- If the snapshot is updated and signed, the `timestamp`(which stores the hash, length,
+ and version of the `snapshot` file) is also updated and signed.
+- If the `root` or a `targets` role is updated, `snapshot`, which stores versions
+of these files, is updated as well.
+- If a target file is updated, information about it in the appropriate targets metadata
+file (`targets` or one corresponding to a delegated role) is updated and signed too.
+
+This all ensures that each revision of the authentication repository constitutes a valid TUF repository.
+Additionally, the version numbers of metadata files in revisions corresponding to subsequent commits
+should either remain the same or differ by one. In essence, this means that only one valid update of the
+TUF repository is permitted per commit. Consequently, all updates to the repository are transparent, and
+all previous versions remain accessible.
+
+Here's an example of how an initial repository might be structured in TAF, with the first set of versioned
+metadata files:
+
+```markdown
+Commit SHA: 1234abcd
+
+- metadata/
+  │
+  ├─ root.json (v1)
+  ├─ targets.json (v1)
+  ├─ snapshot.json (v1)
+  └─ timestamp.json (v1)
+- targets/
+```
+
+Now let's say that `repository.json` is added:
+
+```markdown
+Commit SHA: 5678efgh
+
+- metadata/
+  │
+  ├─ root.json (v1)
+  ├─ targets.json (v2)
+  ├─ snapshot.json (v2)
+  └─ timestamp.json (v2)
+- targets/
+  │
+  └─ repository.json
+```
+
+Finally, let's assume that a new signing key had to be added, which means that `root.json` had
+to be modified and signed:
+
+```markdown
+Commit SHA: 5678efgh
+
+- metadata/
+  │
+  ├─ root.json (v2)
+  ├─ targets.json (v2)
+  ├─ snapshot.json (v3)
+  └─ timestamp.json (v3)
+- targets/
+  │
+  └─ repository.json
+```
+
+To ensure ease of use in managing and maintaining authentication repositories, TAF is equipped with a
+series of Command Line Interface (CLI) commands. These commands are designed to automate the setup
+process, making it accessible and efficient for users. By simply inputting parameters for signing
+keys, signature thresholds, and roles, the CLI commands facilitate the automatic creation of
+authentication repositories. Additionally, for the purpose of updating repositories, TAF's CLI
+commands are tailored to simplify the modification of target and metadata files. This includes tasks
+such as updating the expiration dates of metadata files. The commands also enable the addition of new
+roles, the integration of new target and reference authentication repositories within the TAF
+framework.
```

Comment:
Do we also want to add hyperlink to the[ command line specification](https://github.com/openlawlibrary/taf/blob/master/docs/repo-creation-and-update-util.md)?

Reply:
If we aren't sure about including the command-line specification link here, then I'd vote to remove.